crates/turbopack-core/src/diagnostics/mod.rs

```diff
@@ -0,0 +1,55 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use turbo_tasks::{emit, CollectiblesSource, Upcast, Vc};
+
+/// An arbitrary payload can be used to analyze, diagnose
+/// Turbopack's behavior.
+#[turbo_tasks::value_trait]
+pub trait Diagnostic {
+    // [TODO]: These are subject to change; not finalized yet.
```

Comment:
Can we add docs now, so that we don't forget?

Reply:
Mind elaborate? The comment explains as-is; these interface might change over time while we visiting more telemetry structures (and other diagnostics).

Reply:
Oh I see